eras/babbage/impl/testlib/Test/Cardano/Ledger/Babbage/CDDL.hs

```diff
@@ -0,0 +1,404 @@
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+
+{-# HLINT ignore "Use camelCase" #-}
+{-# HLINT ignore "Evaluate" #-}
+
+module Test.Cardano.Ledger.Babbage.CDDL (
+  module Test.Cardano.Ledger.Alonzo.CDDL,
+  module Test.Cardano.Ledger.Babbage.CDDL,
+) where
+
+import Codec.CBOR.Cuddle.Huddle
+import Data.Function (($))
+import Data.Word (Word64)
+import GHC.Num (Integer)
+import Test.Cardano.Ledger.Alonzo.CDDL hiding (
+  auxiliary_data,
+  block,
+  cost_models,
+  header,
+  header_body,
+  language,
+  major_protocol_version,
+  multiasset,
+  next_major_protocol_version,
+  plutus_data,
+  plutus_script,
+  proposed_protocol_parameter_updates,
+  protocol_param_update,
+  protocol_version,
+  script_data_hash,
+  script_n_of_k,
+  transaction,
+  transaction_body,
+  transaction_output,
+  transaction_witness_set,
+  update,
+ )
+import Text.Heredoc
+
+babbageCDDL :: Huddle
+babbageCDDL = collectFrom [block, transaction, kes_signature, language, signkeyKES]
+
+block :: Rule
+block =
+  comment
+    [str|Valid blocks must also satisfy the following two constraints:
+        |  1) the length of transaction_bodies and transaction_witness_sets must be
+        |     the same
+        |  2) every transaction_index must be strictly smaller than the length of
+        |     transaction_bodies
+        |NEW: 
+        |  invalid_transactions
+        |]
+    $ "block"
+      =:= arr
+        [ a header
+        , "transaction_bodies" ==> arr [0 <+ a transaction_body]
+        , "transaction_witness_sets" ==> arr [0 <+ a transaction_witness_set]
+        , "auxiliary_data_set" ==> mp [0 <+ asKey transaction_index ==> auxiliary_data]
+        , "invalid_transactions" ==> arr [0 <+ a transaction_index]
+        ]
+
+transaction :: Rule
+transaction =
+  "transaction"
+    =:= arr
+      [ a transaction_body
+      , a transaction_witness_set
+      , a VBool
+      , a (auxiliary_data / VNil)
+      ]
+
+header :: Rule
+header = "header" =:= arr [a header_body, "body_signature" ==> kes_signature]
+
+header_body :: Rule
+header_body =
+  comment
+    [str| block_body_size: merkle triple root
+        |      vrf_result: NEW, replaces nonce_vrf and leader_vrf
+        |]
+    $ "header_body"
+      =:= arr
+        [ "block_number" ==> VUInt
+        , "slot" ==> VUInt
+        , "prev_hash" ==> (hash32 / VNil)
+        , "issuer_vkey" ==> vkey
+        , "vrf_vkey" ==> vrf_vkey
+        , "vrf_result" ==> vrf_cert
+        , "block_body_size" ==> VUInt
+        , "block_body_hash" ==> hash32
+        , a operational_cert
```

Comment:
Is this why in the generated cddl, `operational_cert` is defined as  `($kes_vkey, uint, uint, $signature)` instead of how it was before, 
```
  [ hot_vkey        : $kes_vkey
  , sequence_number : uint
  , kes_period      : uint
  , sigma           : $signature
  ]
```

because it is defined in a way that enables reusing it like this, but which atm means that the name of the keys are lost ? Is this right?  Sorry I know we talked about this on the previous PR but I didn't really understand why that was happening, so wondering if my conclusion is correct

Reply:
Yes that is correct. Using an `array []` vs using a `group ()`, we loose keys in the latter. Thanks for bringing this up! This just made me think, we can safely replace `groups` with `arrays` and have no problems with the encoding. I have updated the definition and all the `.cddl` files. Please let me know what you think.

Reply:
It appears that in the case of the `OCert` we can't do this because of the way its CBOR instance is defined. I'm reverting back to the old way, and yes, we will loose the keys for documentation, but I will add them as a comment. :+1: